app/controllers/client2controller.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+class ClientsController extends \BaseController {
+
+	/**
+	 * Display a listing of the resource.
+	 * GET /clients
+	 *
+	 * @return Response
+	 */
+	public function index()
+	{	
+		$pTitle		= "Clients";
+
+		$id 		= Auth::id();
+		$user		= User::find($id);
+		$clients 	= $user->clients()->orderBy('created_at', 'desc')->take(5)->get();
+
+		$counter 	= 0;		
+		return View::make('clients.index', compact([ 'clients', 'counter', 'pTitle']));
+	}
+
+	/**
+	 * Show the form for creating a new resource.
+	 * GET /clients/create
+	 *
+	 * @return Response
+	 */
+	public function create()
+	{
+		$pTitle		= "Clients";
+
+        $name 		= Input::get('name');
+        $user_id 	= Auth::id();
+
+        // Validation
+        $validator = Validator::make(
+            array('name' =>	$name),
+            array('name' => 'required|unique:clients')
+        );
+
+        if ($validator->fails())
+        {
+            return Redirect::back()->withErrors($validator);
+        }
+
+        $client         	= new Client;
+        $client->name   	= $name;
+        $client->user_id 	= $user_id;
+        $client->save();
+
+        return Redirect::back()->with('success', $name ." is now a new client.");
+	}
+
+	/**
+	 * Display the specified resource.
+	 * GET /clients/{id}
+	 *
+	 * @param  int  $id
+	 * @return Response
+	 */
+	public function show($id)
+	{		
+		$client_id  =   $id;
+		$client 	= 	Client::find($id);
+
+		// Bust be refactored as a filter
+		if ( $client->user_id != Auth::id() ) {
+			return Redirect::to('/hud');
+		}
+
+		$projects	=	$client->projects()->get();
+
+		$pTitle		=   $client->name;
+
+		return View::make('clients.show', compact([ 'client', 'projects', 'client_id', 'pTitle' ]));
+	}
+
+	/**
+	 * Show the form for editing the specified resource.
+	 * GET /clients/{id}/edit
+	 *
+	 * @param  int  $id
+	 * @return Response
+	 */
+	public function edit($id)
+	{
+		$client 	= 	Client::find($id);
+		$pTitle		=   "Edit " . $client->name;
+
+		return View::make('clients.edit', compact([ 'client', 'pTitle' ]));
+	}
+
+	/**
+	 * Update the specified resource in storage.
+	 * PUT /clients/{id}
+	 *
+	 * @param  int  $id
+	 * @return Response
+	 */
+	public function update($id)
+	{
+       $name 				= 	Input::get('name');
+       $point_of_contact	=	Input::get('point_of_contact');
+       $phone_number		=	Input::get('phone_number');
+       $email				=	Input::get('email');
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "email" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://172.27.59.144:9000/coding_rules#rule_key=php%3AS1192)